clippy_lints/src/utils/hir_utils.rs

```diff
@@ -512,9 +512,22 @@ impl<'a, 'tcx> SpanlessHash<'a, 'tcx> {
                     self.hash_expr(e);
                 }
             },
-            ExprKind::Tup(ref v) | ExprKind::Array(ref v) => {
+            ExprKind::Tup(ref tup) => {
+                let c: fn(_) -> _ = ExprKind::Tup;
```

Comment:
Please also remove this `fn(..) -> _ ...` line and the two of the match arms below.